src/librustc_borrowck/diagnostics.rs

```diff
@@ -633,6 +633,51 @@ fn main() {
 ```
 "##,
 
+E0503: r##"
+This error occurs when a value is used after mutably borrowing it.
+
+Example of erroneous code:
+
+```compile_fail
+fn main() {
+    let mut value = 3;
+    let _borrow = &mut value;
+    let _sum = value + 1; // error: cannot use `value` because
+                          //        it was mutably borrowed
+}
+```
+
+In this example, `value` is mutably borrowed by `borrow` and cannot be
+used to calculate `sum`. This is not possible because this would violate
+Rust's mutability rules.
+
+You can fix this error by limiting the scope of the borrow:
+
+```
+fn main() {
```

Comment:
Could you annotate this example with some comments to explain what you did please?